staging/src/k8s.io/apiserver/pkg/registry/generic/registry/store.go

```diff
@@ -1365,9 +1367,36 @@ func (e *Store) CompleteWithOptions(options *generic.StoreOptions) error {
 		)
 	}
 
+	if opts.CountMetricPollPeriod > 0 {
+		stopFunc := e.startObservingCount(opts.CountMetricPollPeriod)
+		previousDestroy := e.DestroyFunc
+		e.DestroyFunc = func() {
+			stopFunc()
+			if previousDestroy != nil {
+				previousDestroy()
+			}
+		}
+	}
+
 	return nil
 }
 
+// startObservingCount starts monitoring given prefix and periodically updating metrics. It return a function to stop collection.
+func (e *Store) startObservingCount(period time.Duration) func() {
```

Comment:
What happens for co-habitation? Will we get two monitors?

Reply:
I don't know storage code good enough to answer that. It depends on how co-habitation is implemented and whether two Stores for co-habitating resources share the same Storage object, or control will enter the `if Storage == nil`branch twice.

Having two monitors is not ideal, but it's also not a disaster IMO.

Reply:
Agree. We can live with that.